### ISSUE TITLE
Question about infonce loss

### ISSUE THREAD

**Issue by ksblk2116** [princeton-nlp/SimCSE#212]
In your code, you gather all the embeddings from gpus in  DDP. 
![image](https://user-images.githubusercontent.com/54227312/199640457-3f19a3e9-6ff4-419e-8ead-10aff4a94fb8.png)
But the loss function you comput is not divided by the nums of GPU. I think this [post](https://amsword.medium.com/gradient-backpropagation-with-torch-distributed-all-gather-9f3941a381f8) explain why need to divide.
So,whether the loss is divided by nums of GPU or not doesn't matter? Hope your explaination.Or, I miss something.

**Comment by gaotianyu1350** [comment]
Hi,

We do not need to explicit divide the gradient by the number of GPU here. The loss is divided by the batch size in the end, which takes the number of GPUs into account.
